app/src/main/java/de/danoeh/antennapod/dialog/FeedRefreshIntervalDialog.java

```diff
@@ -0,0 +1,161 @@
+package de.danoeh.antennapod.dialog;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.res.Resources;
+import android.os.Build;
+import android.text.format.DateFormat;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.ArrayAdapter;
+import android.widget.RadioButton;
+import android.widget.RadioGroup;
+import android.widget.Spinner;
+import android.widget.TimePicker;
+
+import androidx.appcompat.app.AlertDialog;
+
+import org.apache.commons.lang3.ArrayUtils;
+
+import java.util.concurrent.TimeUnit;
+
+import de.danoeh.antennapod.R;
+import de.danoeh.antennapod.core.preferences.UserPreferences;
+
+public class FeedRefreshIntervalDialog {
+
+    public interface OnUpdateCallback {
+        void handleUpdate();
+    }
+
+    public static void show(Activity activity, OnUpdateCallback onUpdateCallback) {
+        final Context context = activity;
+        AlertDialog.Builder builder = new AlertDialog.Builder(context);
+        builder.setTitle(R.string.pref_autoUpdateIntervallOrTime_titleAct);
+        builder.setMessage(R.string.pref_autoUpdateIntervallOrTime_sumAct);
+        LayoutInflater inflater = activity.getLayoutInflater();
+
+        View view = inflater.inflate(R.layout.update_refresh_time_dialoxg, null);
+        builder.setView(view);
+
+        final String[] intervalSpinnerValues = context.getResources().getStringArray(R.array.update_intervall_values);
+        final String[] entries = appendHourUnit(activity, intervalSpinnerValues);
```

Comment:
Loading the spinner values from resources feels a bit too complicated to me. How about simply deleting the array there and specifying the values here? Then we also don't need to deal with `String`»`int` conversions.